### PR TITLE
add module graph

### PR DESCRIPTION
**Pull request by jmank88**
This PR adds a `go.md` with a mermaid flowchart of the org module graph for this repo, generated via https://github.com/jmank88/modgraph.
```
go install github.com/jmank88/modgraph@v0.1.0
./modgraph > go.md
```
This is based on the core version:
- https://github.com/smartcontractkit/chainlink/pull/15098

## TODO
- [x] validate from CI

---
**Below is a summarization created by an LLM (gpt-4-0125-preview). Be mindful of hallucinations and verify accuracy.**

## Why
The changes introduce a new GitHub Action workflow for generating module dependency graphs, automate the setup of Go environment for this purpose, and update Go module dependencies. Additionally, they refine the module path corrections in go.mod files to ensure accurate dependency management and introduce a visualization (`go.md`) for module dependencies specific to the project.

## What
- `.github/workflows/modgraph.yaml`  
  Added GitHub Action workflow named `Module graph` that triggers on pull requests, sets up Go, checks out the repository, and generates a module dependency graph ensuring it does not introduce changes.
- `Makefile`  
  Introduced `modgraph` target that installs necessary tools and generates a module dependency visualization (`go.md`) using `modgraph`.
- `framework/examples/myproject/go.mod`  
  Updated module paths for `chainlink-testing-framework` dependencies to correct locations and updated Kubernetes (`k8s.io`) dependencies to `v0.31.2`.
- `framework/examples/myproject/go.sum`  
  Removed unused dependencies, reflecting cleanup and updates in `go.mod`.
- `framework/go.mod`  
  Added a new dependency (`github.com/charmbracelet/huh/spinner`) and updated indirect dependencies reflecting a refinement in project dependencies.
- `framework/go.sum`  
  Updated to reflect changes in `go.mod`, including the addition and removal of various dependencies.
- `go.md`  
  Added a new file to visualize the project's module dependencies in a structured format.
- `modgraph` (script)  
  Introduced a script to generate `go.md`, facilitating the visualization of module dependencies within the project.
- `tools/flakeguard/go.sum`  
  Updated to include a new checksum, possibly as a side effect of changes in the project's dependencies.

### CODE REVIEW

File: framework/examples/myproject/go.mod [hunk 1]
```diff
@@ -3,9 +3,8 @@ module github.com/smartcontractkit/chainlink-testing-framework/framework/example
 go 1.22.7
 
 replace (
-	github.com/smartcontractkit/chainlink-testing-framework/framework => ../..
-	github.com/smartcontractkit/chainlink-testing-framework/framework/cmd => ../../cmd/
-	github.com/smartcontractkit/chainlink-testing-framework/framework/wasp => ../../wasp/
+	github.com/smartcontractkit/chainlink-testing-framework/framework => ../../
+	github.com/smartcontractkit/chainlink-testing-framework/wasp => ../../../wasp
```
Comment (jmank88) [inline framework/examples/myproject/go.mod:7]: These were invalid and tripping up the graph generation, so I fixed them and tidies the whole repo.

Comment (skudasov) [reply framework/examples/myproject/go.mod:7]: That's needed too
```
github.com/smartcontractkit/chainlink-testing-framework/framework/cmd => ../../cmd/
```

Comment (skudasov) [reply framework/examples/myproject/go.mod:7]: If it's blocking your tool I can remove them and stick with tags @jmank88 , these are e2e examples and it's okay to update them if you want to remove replaces at all

Comment (jmank88) [reply framework/examples/myproject/go.mod:7]: What do you mean? There is no `cmd/go.mod` :shrug:

Comment (skudasov) [reply framework/examples/myproject/go.mod:7]: Oh, okay 😄

File: .github/workflows/modgraph.yaml
Comment (jmank88) [inline .github/workflows/modgraph.yaml:1]: I suppose this could use a path filter on `go.mod` files?

**Review by skudasov — APPROVED** [review]

**Pull request merged**